pkg/virt-launcher/virtwrap/cmd-server/client/client.go

```diff
@@ -0,0 +1,238 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package cmdclient
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+/*
+ ATTENTION: Rerun code generators when interface signatures are modified.
+*/
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/rpc"
+	"path/filepath"
+
+	k8sv1 "k8s.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	diskutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type Reply struct {
+	Success        bool
+	Message        string
+	DomainListJSON string
+}
+
+type Args struct {
+	// used for domain management
+	VMJSON          string
+	K8SecretMapJSON string
+
+	// used for syncing secrets
+	SecretUsageType string
+	SecretUsageID   string
+	SecretValue     string
+}
+
+type LauncherClient interface {
+	StartVirtualMachine(vm *v1.VirtualMachine, secrets map[string]*k8sv1.Secret) error
+	ShutdownVirtualMachine(vm *v1.VirtualMachine) error
+	KillVirtualMachine(vm *v1.VirtualMachine) error
+	SyncSecret(vm *v1.VirtualMachine, usageType string, usageID string, secretValue string) error
+	ListDomains() ([]*api.Domain, error)
```

Comment:
Shouldn't that be more something like `GetDomain()(*api.Domain, error)`?

Reply:
I wanted to use the ListAllDomains libvirt api call rather than having the cmd server maintain any state related to the name of the domain it is managing. I could hide that we're using the ListAllDomain's function and just return the first domain entry for the command client's GetDomain function. I didn't see a reason to do that though.

Reply:
That is very confusing, since we know that it can always only ever return one domain (I think we even know the name of the domain to look up).

Reply:
how do we know the name of the domain to look up?  This function is used by the domain informer's List function to sync the cache. There's no knowledge on the client side as to what domain they're requesting at that point.

Reply:
Still, it can just return one domain, or am I at the wrong place in the code?

Reply:
yes, we can hind the fact that the command server is using the ListAllDomain libivrt api function and just call the command client's function GetDomain.  I'm fine with that

Reply:
fixed :)